src/librbd/journal/Types.h

```diff
@@ -121,12 +121,13 @@ struct OpFinishEvent : public OpEventBase {
 
 struct SnapEventBase : public OpEventBase {
   std::string snap_name;
+  cls::rbd::SnapshotNamespace snap_namespace;
```

Comment:
I thought that only user facing functions assume UserSnapshotNamespace by default. Internally we may need to flatten image snapshots, protect and unprotect them. For example when we clone group snapshots into another consistency group). As far as I understand these types are required for those internal operations. Am I wrong?